operation.go

```diff
@@ -84,11 +89,17 @@ func (o *Operation) SuccessResponse() (*Response, int, bool) {
 		return nil, 0, false
 	}
 
-	for k, v := range o.Responses.StatusCodeResponses {
-		if k/100 == 2 {
-			return &v, k, true
+	responseCodes := make([]int, 0, len(o.Responses.StatusCodeResponses))
```

Comment:
Why do you replace this with a more inefficient way of doing it?
the `k/100 == 2` has the exact same effect

Reply:
More inefficient? Mmh interesting point of view. It sure is shorter, but mine gives a clearer intent. About _efficiency_, I am not so sure. Let's get the generated assembly to figure out. Merry Christmas.

Reply:
My version is slightly faster on a x86, by a few nanosecs :)
See [benchmark](https://gist.github.com/fredbi/208f0b95b57ba5763ceb1a482f6212ec).

All this just to say that this is mostly a matter of taste and mind mapping (you think "2xx", I think [200,300[) more than _efficiency_.

The generated assembly (go11, x86 64bit) is a little bit shorter with the `&&` version, with **FIVE** (!!) assembly ops less... Readers will surely take notice of the excellent optimization job carried out with golang here.

Cheers,

F.

Ex:
`return k/100 == 2` 
yields:
```asm
	MOVQ	$-6640827866535438581, AX
	MOVQ	"".k+8(SP), CX
	IMULQ	CX
	ADDQ	CX, DX
	SARQ	$6, DX
	SARQ	$63, CX
	SUBQ	CX, DX
	CMPQ	DX, $2
	SETEQ	"".~r1+16(SP)
	RET
```
Whereas:
`return k>=200 && k<300` 
yields:
```asm
	MOVQ	"".k+8(SP), AX
	ADDQ	$-200, AX
	CMPQ	AX, $100
	SETCS	"".~r1+16(SP)
	RET
```

Reply:
Ah ok, I may explain this (and I should have documented this change).

Well, it is simply that in the case of multiple success responses (like '200', '201', '203'...), the previous version did
randomly return one of these.

The additional complexity is here to ensure that this function is deterministic. Hence the append and sort.